### ISSUE TITLE
gem version 0.0.10 removed from repository

### ISSUE THREAD

**Issue by hampld** [fog/fog-aliyun#8]
Hi, 

is there a specific reason why the version 0.0.10 was completely deleted from repositories? I understand the need for security patches, but deleting the release is kind of inconvenient ...

Best

**Comment by plribeiro3000** [comment]
I don't have permissions. @geemus Can you tackle this please?

**Comment by geemus** [comment]
@hampld I actually have no idea why it was deleted (I didn't do it myself). It may have been in error or by mistake also. @dengqinsi perhaps you know?

**Comment by hampld** [comment]
is there a way to bring this back? This has broken my auto scaling :(

**Comment by sheax0r** [comment]
I actually ran into this earlier today - I had to fix by rolling forward to 0.1.0 :).

**Comment by dengqinsi** [comment]
I deleted version 0.0.10 because it has some bugs. Maybe I should not delete it. I am sorry!
I had pushed 0.1.0, you can use this.

**Comment by dengqinsi** [comment]
@hampld  Unyanking of gems is no longer supported. I don`t know how to bring it back. sorry.

**Comment by plribeiro3000** [comment]
@dengqinsi is OK for a version to have bugs. Só you dont need to yank at all. Just release a new version with a fix asap and tou ae gold.

And btw, take a look at semver.org

**Comment by plribeiro3000** [comment]
Please take a look at how to release a gem as well. Nome of those versions are tagged here. How are you releasing it?

**Comment by hampld** [comment]
Rewriting (git) history is, in my opinion, an absolute no-go. Fog is used by a lot of people in production environments, forcing everyone to hotfix their bundle is not very nice. Unfortunately fog does not allow cherry picking the provider you need ...

To give a little bit of context why this is important (at least to me): No software is perfect, my code certainly is not bug free. It's not something we're proud of as devs, but it is a reality. As soon as complexity exceeds "hello world", there will be bugs. Because we know this, our app is set up in a way to allow individual services to fail. When the service fails to recover after a grace period, the instance is terminated and a new one is spun up automatically. This happened yesterday for us, but the new instance failed to start because your gem was missing from repos. This caused 3 members of my dev-ops team to get messaged at 3AM and one had to get up and patch the gemlock file. Suffice to say, some words were said...

TL;DR don't delete released gems. never ever!

**Comment by dengqinsi** [comment]
@hampld I do apologise! I will never make the same mistake.

**Comment by hampld** [comment]
@dengqinsi then the most important part is already achieved! :)

**Comment by geemus** [comment]
@hampld we are working on the cherry-picking part. Any of the providers that have been separated can be used directly (and we are trying to extract them all, just takes time).

**Comment by geminiyellow** [comment]
不至于吧阿里哥们，你都做 gem 的人了，不要这样害同行好吧。

**Comment by plribeiro3000** [comment]
@geminiyellow could you write in english plz? I'm not sure if i translated this correctly.

Thanks!

**Comment by plribeiro3000** [comment]
@dengqinsi Are you still yanking gem versions? It seems this is still happening.
